xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -370,33 +381,108 @@ send_fuse_iov(xlator_t *this, fuse_in_header_t *finh, struct iovec *iov_out,
         fouh->len += iov_out[i].iov_len;
     fouh->unique = finh->unique;
 
-    res = sys_writev(priv->fd, iov_out, count);
-    gf_log("glusterfs-fuse", GF_LOG_TRACE, "writev() result %d/%d %s", res,
-           fouh->len, res == -1 ? strerror(errno) : "");
+    if (sync) {
```

Comment:
In what case this is needed ? doing sync writes blocks worker threads, which is what we are trying to avoid with io_uring.

Reply:
In case send_fuse_err(), keeping in sync. When there is no memory and fuse wants to send err to /dev/fuse, sending the error in async mode required memory to be allocated which will again fail. So preferring sync mode for error case